src/Symfony/Component/Messenger/Bridge/AmazonSqs/Tests/Transport/ConnectionTest.php

```diff
@@ -115,6 +115,24 @@ public function testFromDsnWithQueryOptions()
         );
     }
 
+    public function testFromDsnWithQueueNameOption()
+    {
+        $httpClient = $this->getMockBuilder(HttpClientInterface::class)->getMock();
+        $this->assertEquals(
+            new Connection(['queue_name' => 'queue'], new SqsClient(['region' => 'eu-west-1', 'accessKeyId' => null, 'accessKeySecret' => null], null, $httpClient)),
+            Connection::fromDsn('sqs://default', ['queue_name' => 'queue'], $httpClient)
+        );
+    }
+
+    public function testFromDsnWithOverridingQueueNameByOption()
+    {
+        $httpClient = $this->getMockBuilder(HttpClientInterface::class)->getMock();
+        $this->assertEquals(
+            new Connection(['queue_name' => 'overrided_queue'], new SqsClient(['region' => 'eu-west-1', 'accessKeyId' => null, 'accessKeySecret' => null], null, $httpClient)),
```

Comment:
I agree with previous test but this one is wrong. For consistency with all other Symfony components: DSN must take precedence over options.